flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/sink/writer/PulsarWriter.java

```diff
@@ -244,4 +261,31 @@ public void close() throws Exception {
         // Close all the resources and throw the exception at last.
         closeAll(metadataListener, producerRegister);
     }
+
+    private void setupFlinkMetrics(PulsarSinkWriterMetrics pulsarSinkWriterMetrics) {
+        registerGauges();
+        registerMetricUpdateTimer();
+    }
+
+    private void registerGauges() {
+        pulsarSinkWriterMetrics.setCurrentSendTimeGauge(this::getCurrentSendTime);
```

Comment:
I am experiement with this, if I write
```
pulsarSinkWriterMetrics.setCurrentSendTimeGauge(() -> lastSendTime);
```
I remember in the old past, variables in lambdas must be final or effectively final, but here lastSendTime is modified in the code, but IDE shows no warning. I'm curious if I write
```
pulsarSinkWriterMetrics.setCurrentSendTimeGauge(() -> lastSendTime);
```
Will lastSendTime always be the same value or it can show the updates ?